### PR TITLE
Use the original TargetFramework string for conditions in nuget.g.* for frameworks with platforms (such as net5.0-windows)

### LINKED ISSUE #9914: Projects targetting net5.0-windows, end up generating a condition with an incorrect TargetFramework string
This is a consequence of https://github.com/nuget/home/issues/9756. 

Some initial work started in `dev-nkolev92-originalTargetFrameworkCondition`

I'd estimate a 5-10 point task for someone familiar with the problem space.

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/9914
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: 

This is a consequence of #9756.
The idea is simple, earlier we'd match based on which tfm string parses to the nugetframework, now we look at the alias of TargetFrameworkInformation.

## Testing/Validation

Tests Added: Yes  
Reason for not adding tests:  
Validation:

### CODE REVIEW

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs [hunk 1]
```diff
@@ -227,6 +231,11 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
             return 0;
         }
 
+        private static string GetMatchingFrameworkStrings(PackageSpec spec, NuGetFramework framework)
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: GetAliasForFramework?

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs [hunk 2]
```diff
@@ -209,7 +209,11 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
                     packageReferenceArgs.PackageId,
                     packageReferenceArgs.ProjectPath));
 
-                var compatibleOriginalFrameworks = originalPackageSpec.RestoreMetadata
+                var compatibleOriginalFrameworks = compatibleFrameworks
+                    .Select(e => GetMatchingFrameworkStrings(originalPackageSpec, e))
+                    .Where(originalFramework => originalFramework != null);
+
+                    originalPackageSpec.RestoreMetadata
```
Comment (erdembayar) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: I don't understand what is going on here? You filtered something but not assigned to anything, so it'll be lost.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: It's used in a line that I did not change, so it won't show up in the diff. 
See https://github.com/NuGet/NuGet.Client/blob/e1fb834826d4ee8203dcefda767df107eb25e716/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs#L219-L221

Comment (zivkan) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: I think @erdembayar's right here. Line 214 ends in a semicolon, and this line starts a new LINQ query, but the result is not used anywhere, and therefore will never be evaluated.

Comment (erdembayar) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: Yes, I think you need to replace replace `;` with `.` in Line 214.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: It took me a while to get what the feedback was. 

No need for `.`. 
The code I added replaces that. That's dead code. 

Removed.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: fixed.

**Review by erdembayar — APPROVED** [review]

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs [hunk 3]
```diff
@@ -10009,6 +10007,60 @@ public void RestoreNetCore_WithMultipleFrameworksWithPlatformAndAssetTargetFallb
...
+                    "net50");
+
+                // Workaround: Set all the TFM properties ourselves.
+                // We can't rely on the SDK setting them, as only .NET 5 SDK P8 and later applies these correctly.
+                var net50windowsTFM = project.Frameworks.Where(f => f.TargetAlias.Equals("net5.0-windows")).Single();
+                var net50TFM = project.Frameworks.Where(f => f.TargetAlias.Equals("net50")).Single();
+                net50windowsTFM.Properties.Add("TargetFrameworkMoniker", ".NETCoreApp, Version=v5.0");
+                net50windowsTFM.Properties.Add("TargetPlatformMoniker", "Windows, Version=7.0");
+                net50TFM.Properties.Add("TargetFrameworkMoniker", ".NETCoreApp, Version=v5.0");
+                net50TFM.Properties.Add("TargetPlatformMoniker", " ,Version= ");
```
Comment (zivkan) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs]: " ,Version= " seems really weird, and in my opinion distracts from what the point of this test is supposed to be testing. Doesn't the code handle the case when TPM is not defined, or at worst case `string.Empty`?

**Review by zivkan — APPROVED** [review]

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs (hunk 3 above)
Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs]: The current SDK we have for testing sets a default and makes my test cry with a double tfm problem. 
So the problem is that the prop is defined right now.

Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs]: I switched to android as a test. No net5 only test, but that's ok :)

**Pull request merged**